src/template/template.js

```diff
@@ -0,0 +1,55 @@
+const path = require('path');
+const fs = require('fs-extra-promise');
+
+const requiredFiles = ['index.md', 'site.json', '_markbind/'];
+
+function Template(rootPath, templatePath) {
+  this.rootPath = rootPath;
+  this.template = templatePath;
+}
+
+function validateTemplateFromPath(templatePath) {
+  for (let i = 0; i < requiredFiles.length; i += 1) {
+    const requiredFile = requiredFiles[i];
+    const requiredFilePath = path.join(templatePath, requiredFile);
+
+    if (!fs.existsSync(requiredFilePath)) {
+      return false;
+    }
+  }
+
+  return true;
+}
+
+function generateSiteWithTemplate(rootPath, templatePath) {
+  return new Promise((resolve, reject) => {
+    fs.accessAsync(rootPath)
+      .catch(() => fs.emptydirSync(rootPath))
```

Comment:
I don't think it is appropriate to wipe the rootPath directory should generation fail, it may already have some existing content before MarkBind was invoked, and it is not a good idea to delete those files.

Reply:
I thought that was the behaviour, it creates the directory if accessAsync fails as it assumes that the directory does not exist.

Reply:
Perhaps `fs.mkdirSync()` would be more appropriate for implementing that behaviour instead?